pkg/occlient/occlient.go

```diff
@@ -2212,15 +2212,15 @@ func (c *Client) CreateServiceInstance(serviceName string, serviceType string, s
 
 // CreateServiceBinding creates a ServiceBinding (essentially a secret) within the namespace of the
 // service instance created using the service's parameters.
-func (c *Client) CreateServiceBinding(bindingName string, namespace string) error {
+func (c *Client) CreateServiceBinding(bindingName string, namespace string, labels map[string]string) error {
```

Comment:
Shouldn't this throw an error when the bindingName or namespace are empty strings?

Reply:
I think so. I think it should throw an error for empty value of `namespace` as well.

Reply:
Yeah I think so. It shouldn't be a empty value right?

Reply:
> I think so. I think it should throw an error for empty value of `namespace` as well.

He is talking about ```bindingName```. I am also thinking about ```labels```, shouldn't this will throw error for empty labels?

Reply:
So, just for the sake of it, I changed the values of `bindingName` and `namespace` to `""` before the call to create a ServiceBinding. And it failed with below errors:
```sh
# when namespace is ""
odo service create dh-postgresql-apb psql-1 --plan dev -p postgresql_database=mydb -p postgresql_password=password -p postgresql_user=luke -p postgresql_version=10
Deploying service psql-1 of type: dh-postgresql-apb
 ✗  Deploying service [369ms]
 ✗  an empty namespace may not be set during creation

# when bindingNs is ""
$ odo service create dh-postgresql-apb psql-1 --plan dev -p postgresql_database=mydb -p postgresql_password=password -p postgresql_user=luke -p postgresql_version=10
Deploying service psql-1 of type: dh-postgresql-apb
 ✗  Deploying service [564ms]
 ✗  ServiceBinding.servicecatalog.k8s.io "" is invalid: [metadata.name: Required value: name or generateName is required, spec.instanceRef.name: Invalid value: "": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '
-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.secretName: Invalid value: "": a DNS-1123 subdomai
n must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')]
```

What's surprising in both cases is that in spite of the error, service gets deployed and provisioned:
```sh
$ odo service list --app app
NAME       TYPE                  PLAN     STATUS
psql-1     dh-postgresql-apb     dev      ProvisionedSuccessfully
```

But, again in both the cases, the service deletion fails since it can't find the corresponding service binding:
```sh
$ odo service delete psql-1 --app=app -f
 ✗  unable to delete service psql-1:
unable to delete serviceBinding: servicebindings.servicecatalog.k8s.io "psql-1" not found
```

So one needs to do `oc delete` to delete the ServiceInstance.

Reply:
> I am also thinking about labels, shouldn't this will throw error for empty labels?

Empty labels are fine because they are not mandatory to create an API object. I tried it out btw.

Reply:
> So, just for the sake of it, I changed the values of bindingName and namespace to "" before the call to create a ServiceBinding. And it failed with below errors:

So I guess bindingName and namespace shouldn't be empty and thus we should have a check for that in this function to make sure that we don't face the described error.

Reply:
Opened #2493 to handle this. Is that OK @mik-dass? Those lines are unrelated to this PR and I'd like to keep it that way.

Reply:
ok :+1:

Reply:
I just added a [comment](https://github.com/openshift/odo/issues/2493#issuecomment-572023557) on #2493  related to this.